app/assets/javascripts/oxalis/model/reducers/volumetracing_reducer.js

```diff
@@ -62,16 +62,16 @@ function VolumeTracingReducer(state: OxalisState, action: VolumeTracingActionTyp
   return getVolumeTracing(state.tracing)
     .map(volumeTracing => {
       switch (action.type) {
-        case "SET_MODE": {
-          return setModeReducer(state, volumeTracing, action.mode);
+        case "SET_TOOL": {
+          return setToolReducer(state, volumeTracing, action.tool);
         }
 
-        case "TOGGLE_MODE": {
-          const newMode =
-            volumeTracing.volumeTraceOrMoveMode === Constants.VOLUME_MODE_TRACE
-              ? Constants.VOLUME_MODE_MOVE
-              : Constants.VOLUME_MODE_TRACE;
-          return setModeReducer(state, volumeTracing, newMode);
+        case "TOGGLE_TOOL": {
```

Comment:
Can we rename this to `CYCLE_TOOL` or something similar? I'd use "toggle" only in a binary context.